### PR TITLE
ci(engine): fix duplicated checkpoint in fake job

### LINKED ISSUE #5692: node failure chaos case failed
### What did you do?

https://github.com/pingcap/tiflow/runs/6679948879?check_suite_focus=true

### What did you expect to see?

case runs successfully

### What did you see instead?

We have duplicated worker status

```go
executor-3.log:[2022/05/31 23:18:16.693 +00:00] [INFO] [fake_master.go:321] ["FakeMaster: Tick"] [status="{\"01ffbac6-3fa9-445d-ac00-a73b26f47e86\":{\"business-id\":5,\"tick\":10296,\"checkpoint\":{\"tick\":0,\"revision\":1429,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"0540b2fc-af29-40d5-a475-d9fbe7700339\":{\"business-id\":0,\"tick\":2701,\"checkpoint\":{\"tick\":0,\"revision\":951,\"mvcc-count\":3,\"value\":\"update-value-index-2\"}},\"279f3051-208e-45e2-b6df-66c81dec2b5e\":{\"business-id\":0,\"tick\":9878,\"checkpoint\":{\"tick\":0,\"revision\":1424,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"3181411d-871c-4681-9839-ab9c1beeb35e\":{\"business-id\":2,\"tick\":10714,\"checkpoint\":{\"tick\":0,\"revision\":1426,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"59e76b23-afc3-44d8-a59c-1bf7efd6fb72\":{\"business-id\":6,\"tick\":10296,\"checkpoint\":{\"tick\":0,\"revision\":1430,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"64c62b4c-d9c2-4c0c-88a3-5452e934b24e\":{\"business-id\":1,\"tick\":317,\"checkpoint\":{\"tick\":0,\"revision\":4,\"mvcc-count\":1,\"value\":\"update-value-index-0\"}},\"6a3f3d55-1b53-4cda-8628-54d330c52e3e\":{\"business-id\":7,\"tick\":317,\"checkpoint\":{\"tick\":0,\"revision\":10,\"mvcc-count\":1,\"value\":\"update-value-index-0\"}},\"840b8940-3baf-42b2-ae17-74248016b20b\":{\"business-id\":1,\"tick\":9498,\"checkpoint\":{\"tick\":0,\"revision\":1425,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"86c2f62c-091f-4750-8221-bb18079787a8\":{\"business-id\":6,\"tick\":2701,\"checkpoint\":{\"tick\":0,\"revision\":957,\"mvcc-count\":3,\"value\":\"update-value-index-2\"}},\"a0f0361a-962c-42ad-b073-aa9511745df8\":{\"business-id\":4,\"tick\":10712,\"checkpoint\":{\"tick\":0,\"revision\":1428,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"acad1ccd-8a69-4457-adce-07d362390b07\":{\"business-id\":0,\"tick\":9484,\"checkpoint\":{\"tick\":0,\"revision\":1424,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"d280617c-b8a4-45f2-9535-dbcfbc0c8b7c\":{\"business-id\":3,\"tick\":10295,\"checkpoint\":{\"tick\":0,\"revision\":1427,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"d737889f-188d-4122-975e-5d73161a547f\":{\"business-id\":1,\"tick\":9892,\"checkpoint\":{\"tick\":0,\"revision\":1425,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"da11cd0a-a86c-43ad-bab8-89d03e2dcc71\":{\"business-id\":3,\"tick\":9902,\"checkpoint\":{\"tick\":0,\"revision\":1427,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"f643dced-3fbf-4fe7-855e-a357fdba2921\":{\"business-id\":5,\"tick\":9901,\"checkpoint\":{\"tick\":0,\"revision\":1429,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}},\"fe57e88b-5014-433b-abfc-381d9c5f8dad\":{\"business-id\":7,\"tick\":10311,\"checkpoint\":{\"tick\":0,\"revision\":1431,\"mvcc-count\":4,\"value\":\"update-value-index-3\"}}}"]
```

Especially for status

```json
{
   "01ffbac6-3fa9-445d-ac00-a73b26f47e86": {
      "business-id": 5,
      "tick": 10296,
      "checkpoint": {
         "tick": 0,
         "revision": 1429,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "0540b2fc-af29-40d5-a475-d9fbe7700339": {
      "business-id": 0,
      "tick": 2701,
      "checkpoint": {
         "tick": 0,
         "revision": 951,
         "mvcc-count": 3,
         "value": "update-value-index-2"
      }
   },
   "279f3051-208e-45e2-b6df-66c81dec2b5e": {
      "business-id": 0,
      "tick": 9878,
      "checkpoint": {
         "tick": 0,
         "revision": 1424,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "3181411d-871c-4681-9839-ab9c1beeb35e": {
      "business-id": 2,
      "tick": 10714,
      "checkpoint": {
         "tick": 0,
         "revision": 1426,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "59e76b23-afc3-44d8-a59c-1bf7efd6fb72": {
      "business-id": 6,
      "tick": 10296,
      "checkpoint": {
         "tick": 0,
         "revision": 1430,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "64c62b4c-d9c2-4c0c-88a3-5452e934b24e": {
      "business-id": 1,
      "tick": 317,
      "checkpoint": {
         "tick": 0,
         "revision": 4,
         "mvcc-count": 1,
         "value": "update-value-index-0"
      }
   },
   "6a3f3d55-1b53-4cda-8628-54d330c52e3e": {
      "business-id": 7,
      "tick": 317,
      "checkpoint": {
         "tick": 0,
         "revision": 10,
         "mvcc-count": 1,
         "value": "update-value-index-0"
      }
   },
   "840b8940-3baf-42b2-ae17-74248016b20b": {
      "business-id": 1,
      "tick": 9498,
      "checkpoint": {
         "tick": 0,
         "revision": 1425,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "86c2f62c-091f-4750-8221-bb18079787a8": {
      "business-id": 6,
      "tick": 2701,
      "checkpoint": {
         "tick": 0,
         "revision": 957,
         "mvcc-count": 3,
         "value": "update-value-index-2"
      }
   },
   "a0f0361a-962c-42ad-b073-aa9511745df8": {
      "business-id": 4,
      "tick": 10712,
      "checkpoint": {
         "tick": 0,
         "revision": 1428,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "acad1ccd-8a69-4457-adce-07d362390b07": {
      "business-id": 0,
      "tick": 9484,
      "checkpoint": {
         "tick": 0,
         "revision": 1424,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "d280617c-b8a4-45f2-9535-dbcfbc0c8b7c": {
      "business-id": 3,
      "tick": 10295,
      "checkpoint": {
         "tick": 0,
         "revision": 1427,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "d737889f-188d-4122-975e-5d73161a547f": {
      "business-id": 1,
      "tick": 9892,
      "checkpoint": {
         "tick": 0,
         "revision": 1425,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "da11cd0a-a86c-43ad-bab8-89d03e2dcc71": {
      "business-id": 3,
      "tick": 9902,
      "checkpoint": {
         "tick": 0,
         "revision": 1427,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "f643dced-3fbf-4fe7-855e-a357fdba2921": {
      "business-id": 5,
      "tick": 9901,
      "checkpoint": {
         "tick": 0,
         "revision": 1429,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   },
   "fe57e88b-5014-433b-abfc-381d9c5f8dad": {
      "business-id": 7,
      "tick": 10311,
      "checkpoint": {
         "tick": 0,
         "revision": 1431,
         "mvcc-count": 4,
         "value": "update-value-index-3"
      }
   }
}
```

### Versions of the cluster

Dataflow Engine version:

master@https://github.com/pingcap/tiflow/commit/586885255c1aef5800cde8ede527a575d49a6604

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: close #5692 

### What is changed and how it works?

- Remove failed worker handler when processing `OnWorkerOffline`

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: .github/workflows/dataflow_engine_chaos.yaml [hunk 1]
```diff
@@ -3,6 +3,8 @@ name: Dataflow Engine Chaos
 on:
   schedule:
     - cron: '0 17-23 * * *' # run at minute 0 every hour from 01:00 ~ 07:00 UTC+8
+  pull_request:
+    branches: [ master ]
```
Comment (amyangfei) [inline .github/workflows/dataflow_engine_chaos.yaml:7]: Trigger chaos test, remove this change before PR is merged.

File: engine/lib/fake/fake_master.go [hunk 2]
```diff
@@ -239,17 +239,21 @@ func (m *Master) tickedCheckWorkers(ctx context.Context) error {
 			return nil
 		}
 		m.initialized = true
+
+		// clean tombstone workers, add active workers
 		for _, worker := range m.GetWorkers() {
 			if worker.GetTombstone() != nil {
+				if err := worker.GetTombstone().CleanTombstone(ctx); err != nil {
```
Comment (maxshuang) [inline engine/lib/fake/fake_master.go:246]: This name sounds strange. It sounds like `the worker clean itself`.

**Pull request merged**